### PR TITLE
MacOS Monterey support

### PR DESCRIPTION
**Pull request by a-hromov**
Fixed things that were making problems during build on MacOS Monterey 12.0.1

CodeLLDB extension of Visual Studio Code has been used as a debugger

### CODE REVIEW

**Comment by TohaKi11** [comment]
I also updated the wiki page with installation instructions, since they were obsolete.

Please take a look and if they're fine - please clone them into the main repo:
https://github.com/TohaKi11/OpenImageDebugger/wiki/Building-on-MacOS

**Comment by brunoalr** [comment]
Great PR, thanks! I'm reviewing it. So far I have mostly minor nitpicks. I wonder why Travis was not triggered :thinking:.

**Comment by TohaKi11** [comment]
It is worth triggering CI, since there are a huge amount of CMake changes

**Comment by brunoalr** [comment]
> It is worth triggering CI, since there are a huge amount of CMake changes

Yes, I'm trying to check what's going wrong.

**Comment by brunoalr** [comment]
Hi @TohaKi11 I'm currently on holidays without easy access to a laptop, so I cannot finish the review soon. In the meantime I pinged @csantosbh, I hope that he's not too busy now :)

**Review by brunoalr — CHANGES_REQUESTED** [review]

**Comment by brunoalr** [comment]
I also published the initial minor nits I had. The main point is not there though. Basically I'm not sure if I like the idea of defaulting to error neither the idea of using a specific flag for CI. I'm open for discussions later on l.

File: common.cmake [hunk 1]
```diff
@@ -23,8 +23,14 @@
...
 
+if(NOT APPLE)
+    set(CMAKE_INSTALL_PREFIX_DEFAULT /usr/local)
+else()
+    set(CMAKE_INSTALL_PREFIX_DEFAULT /usr/local/bin)
+endif()
+
 set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type")
-set(CMAKE_INSTALL_PREFIX /usr/local CACHE PATH "Install path")
+set(CMAKE_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX_DEFAULT} CACHE PATH "Install path" FORCE)
```
[the remark this answers is not in the document]
Comment (TohaKi11) [reply common.cmake:33]: Maybe that's something that was left by mistake, I will remove that

[the remark this answers is not in the document]
Comment (TohaKi11) [reply common.cmake]: I double checked the behavior with and without FORCE and here are my findings:
- CMake sets default CMAKE_INSTALL_PREFIX value before reading CMakeLists.txt. On my machine it sets that to /usr/local
- When CMake reads this line in CMakeLists.txt it already has a value behind a variable and don't modifies it
- FORCE is needed to modify it even in that case
- But without additional check FORCE will be just changing any user input
- So, it is needed to be wrapped with CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT check
- See more details here: https://stackoverflow.com/a/39485990

Similar situation is with CMAKE_BUILD_TYPE variable. the solution is here:https://stackoverflow.com/a/48832234

**Comment by TohaKi11** [comment]
Hi @brunoalr and @csantosbh 

I've recently switched my working IDE to CLion and I see that the story with python library complexity continues.

CLion's lldb debugger uses `Python.framework` instead of `libpython*.dylib`.
In order to support that I added one more parameter for the user's input - `APPLE_LLDB_PYTHON3_FRAMEWORK_DIR`.

Now the user has two options: python library file OR framework directory.

Please check the PR and here is a link to the updated wiki page, feel free to cherry-pick it as well:
https://github.com/TohaKi11/OpenImageDebugger/wiki/Building-on-MacOS

Happy holidays,
Anton

**Comment by brunoalr** [comment]
Hi @TohaKi11 I haven't forgotten you. It's just that life is a bit too busy lately. I will come back to you as soon as it is possible :) Thanks for your enthusiastic contributions! :D

**Comment by brunoalr** [comment]
closing and reopening to trigger new CI

**Comment by brunoalr** [comment]
after a long time, I came back :) I cherry-picked some commits that touched the code itself and I am still trying to figure out how to automatically ensure that OID is linked to the same python binaries as the ones used by lldb

**Pull request closed without merging**